File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/SerializedChatMessage.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class SerializedChatMessage(
  val contextFiles: List<ContextItem>? = null,
  val error: ChatError? = null,
  val editorState: Any? = null,
  val speaker: SpeakerEnum, // Oneof: human, assistant, system
  val text: String? = null,
  val model: String? = null,
) {

  enum class SpeakerEnum {
    @SerializedName("human") Human,
    @SerializedName("assistant") Assistant,
    @SerializedName("system") System,
  }
}

File: agent/bindings/csharp/SerializedChatMessage.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class SerializedChatMessage
  {
    [JsonProperty(PropertyName = "contextFiles")]
    public ContextItem[] ContextFiles { get; set; }
    [JsonProperty(PropertyName = "error")]
    public ChatError Error { get; set; }
    [JsonProperty(PropertyName = "editorState")]
    public Object EditorState { get; set; }
    [JsonProperty(PropertyName = "speaker")]
    public SpeakerEnum Speaker { get; set; } // Oneof: human, assistant, system
    [JsonProperty(PropertyName = "text")]
    public string Text { get; set; }
    [JsonProperty(PropertyName = "model")]
    public string Model { get; set; }

    public enum SpeakerEnum
    {
      [EnumMember(Value = "human")]
      Human,
      [EnumMember(Value = "assistant")]
      Assistant,
      [EnumMember(Value = "system")]
      System,
    }
  }
}
